cmd/cmd.go

```diff
@@ -369,6 +414,20 @@ func PushHandler(cmd *cobra.Command, args []string) error {
 
 	request := api.PushRequest{Name: args[0], Insecure: insecure}
 	if err := client.Push(cmd.Context(), &request, fn); err != nil {
+		if spinner != nil {
+			spinner.Stop()
+		}
+		if strings.Contains(err.Error(), "access denied") {
+			return errors.New("you are not authorized to push to this namespace, create the model under a namespace you own")
+		}
+		host := model.ParseName(args[0]).Host()
+		isOllamaHost := host == model.DefaultHost || strings.HasSuffix(host, "ollama.ai") || strings.HasSuffix(host, "ollama.com")
```

Comment:
Why does this match on DefaultHost if it's already matching the suffix? Also, this suffix matching will match other domains that may end in `ollama.com` e.g. `exampleollama.com`. If you're matching a subdomain, it's better to match `.ollama.com`